runtime/caml/config_compat.h

```diff
@@ -2,26 +2,34 @@
 /*                                                                        */
 /*                                 OCaml                                  */
 /*                                                                        */
-/*                  Jeremie Dimino, Jane Street Group, LLC                */
+/*                         Antonin Decimo, Tarides                        */
 /*                                                                        */
-/*   Copyright 2015 Institut National de Recherche en Informatique et     */
-/*     en Automatique.                                                    */
+/*   Copyright 2024 Tarides                                               */
 /*                                                                        */
 /*   All rights reserved.  This file is distributed under the terms of    */
 /*   the GNU Lesser General Public License version 2.1, with the          */
 /*   special exception on linking described in the file LICENSE.          */
 /*                                                                        */
 /**************************************************************************/
 
-/* This file is used by the configure test program nanosecond_stat.c
-   and stat.c in this directory */
+/* Definitions for compatibility with old identifiers. */
 
-#if HAS_NANOSECOND_STAT == 1
-#  define NSEC(buf, field) buf->st_##field##tim.tv_nsec
-#elif HAS_NANOSECOND_STAT == 2
-#  define NSEC(buf, field) buf->st_##field##timespec.tv_nsec
-#elif HAS_NANOSECOND_STAT == 3
-#  define NSEC(buf, field) buf->st_##field##timensec
-#else
-#  define NSEC(buf, field) 0
+#ifndef CAML_CONFIG_COMPAT_H
+#define CAML_CONFIG_COMPAT_H
+
+#ifndef CAML_INTERNALS
```

Comment:
We can't co-opt `CAML_INTERNALS` this way, I'm afraid. Defining `CAML_INTERNALS` means you've opted _in_ to the internals; its negation does not mean that we're building OCaml itself!

FWIW I wouldn't worry about guarding this against internal usage, so I'd just unconditionally define them, as before.